conda_forge_tick/mamba_solver.py

```diff
@@ -297,8 +357,52 @@ def solve(self, specs) -> Tuple[bool, List[str]]:
                 solver.problems_to_str(),
             )
             err = solver.problems_to_str()
+            solution = None
+            run_exports = {"weak": set(), "strong": set()}
+        else:
+            t = api.Transaction(solver, PACKAGE_CACHE)
+            solution = []
+            _, to_link, _ = t.to_conda()
+            for _, _, jdata in to_link:
+                data = json.loads(jdata)
+                solution.append(
+                    " ".join([data["name"], data["version"], data["build"]]),
+                )
+
+            if get_run_exports:
+                run_exports = self._get_run_exports(to_link)
+
+        if get_run_exports:
```

Comment:
I'm not over the moon about having a variable length return. It may be the only way forward, but it does produce some extra cognitive load

Reply:
Yeah me neither but this is the most efficient thing.